feat: pact broker migration docs PACT-158

Inksprout opened the pull request:
This PR adds a detailed section on how to migrate existing Pact Broker db for on-prem users. I have also added a brief overview of the process to the 'Getting Started' section for SAAS users, who will need to contact us to do this as part of their onboarding.

We will need to follow up with some process notes on how to do this for our own reference for SAAS users.

I'll also update the customer care team issue index to flag this with them, any tickets with this type of query will need to be assigned to us.

YOU54F commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,34 @@
+
+## Single Pact Broker instance
+
+Follow existing [set up documentation](https://docs.pactflow.io/docs/on-premises) for setting up your new Pactflow, until the section titled ['Database'](https://docs.pactflow.io/docs/on-premises/database).
```
typo possibly - under the section titled?

Inksprout replied:
sorry where?

YOU54F replied:
Oh I see thanks for the screenshot super helpful.

In that case at the top of the database step, you could have a note there - if you are migrating an existing database skip this section and go straight to the migration page.

YOU54F commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,34 @@
+
+To migrate your existing Pact Broker follow these steps instead of creating a new database:
+
+1. Re-use your existing Pact Broker environment variables, updating the names to read `PACTFLOW_` instead of `PACT_BROKER_`. It is crucial that the values for `DATABASE` related environment variables remain the same so that your Pact Broker database will be updated and used. A full list of the Pactflow environment variables can be found [here](https://docs.pactflow.io/docs/on-premises/environment ...
```
Is it worth calling out the pattern for the env vars `*_DATABASE_*`

I assume a user needs to both

rename all 

`PACT_BROKER_DATABASE_*

to 

`PACTFLOW_DATABASE_*

I would maybe be use italicising or use a caution note feature of Docusaurus. [reference](https://docusaurus.io/docs/next/markdown-features/admonitions)

> existing Pact Broker database will be used, and updated for use with Pactflow as part of the process.

Just some copy change to make it clear we are updating the users existing database.

YOU54F reviewed (approved):
Great addition, added a couple of comments. 

Will test it out and refine if necessary

YOU54F commented on website/docs/docs/getting-started.mdx:
note - applies to Saas (Hosted) Pactflow Brokers only.

Using On-Prem and want to migrate? Check out our on-prem migration guide [here](https://github.com/pactflow/docs.pactflow.io/pull/139/files#diff-420e4d0501183c162474f32b8d736d707f7eb18ad5e8391c7bec5c13f122f44b)

bethesque replied:
Yes, worth clarifying "When you sign up". Maybe "When you sign up to the SaaS Pactflow platform..."

YOU54F commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,34 @@
+
+# Migrating from Pact Broker
+
+It is possible to migrate from using the open source Pact Broker to Pactflow, keeping all existing data and making it available in Pactflow. The migration process uses the existing Pact Broker database, database migrations are then performed to add additional tables and data used for Pactflow features.
```
Assume Postgres db’s only for migration.

should we state that, as I believe support has dropped for MySQL?

https://github.com/pact-foundation/pact_broker#rolling-your-own

mefellows replied:
Ah yes, this is a good call out. We'll want a DB dump from Postgres.

Perhaps we just add one of those note sections e.g.

```
:::note
Migration is only supported for a Pact Broker running a Postgres database.
:::
```

Inksprout replied:
Good idea, yeah I interpret the same thing based on those docs.

Inksprout replied:
@mefellows would there be any other dbs supported? Or just Postgres? I know that when I run pactflow locally it's set up as an sqlite db🤔. wondering if people would use that or anything else we could support when using Pact Broker?

mefellows replied:
I don't think we could reliably import an sqlite DB. We certainly don't recommend running it in "real life" (although people do!).

bethesque replied:
Agreed.

bethesque commented on website/docs/docs/getting-started.mdx:
I'd say "accounts" rather than "Brokers". And "SaaS".

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+
+# Migrating from Pact Broker
+
+If you have previously used the open source Pact Broker with a PostgreSQL database it is possible to migrate that data to Pactflow, keeping everything and making it available in Pactflow. The migration process uses the existing Pact Broker database, database migrations are then performed to add additional tables and data used for Pactflow features.
```
I'm a grammar pedant. This is called comma splicing, and I never realised what it was until I had a friend who is an even worse grammar pedant than me call me out out on it multiple times. "The migration process uses the existing Pact Broker database, database migrations are then performed to add additional tables and data used for Pactflow features." should either be two sentences, or have an "and" after the comma.

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+
+:::note
+
+It is crucial that all `PACTBROKER_DATABASE_*` environment variables are renamed to `PACTFLOW_DATABASE_*` while their `values` remain the same. This is because your existing Pact Broker database will be used, and updated for use with Pactflow as part of the process.
```
```suggestion
`PACT_BROKER_DATABASE_*`
```

bethesque replied:
Should be `PACT_BROKER_DATABASE_*`

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+:::
+
+
+2. Remove the environment variables called `PACT_BROKER_AUTO_MIGRATE_DB` and `PACT_BROKER_AUTO_MIGRATE_DB_DATA`
```
Qualify this with "if you have them set". They should not add them if they weren't already using them (99% of people won't be using them).

Inksprout replied:
Just to clarify: If they did not previously use PACT_BROKER_AUTO_MIGRATE_DB and PACT_BROKER_AUTO_MIGRATE_DB_DATA, then they should not add the 'PACTFLOW_DATABASE_AUTO_MIGRATE=true' env var?

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+
+
+2. Remove the environment variables called `PACT_BROKER_AUTO_MIGRATE_DB` and `PACT_BROKER_AUTO_MIGRATE_DB_DATA`
+3. Add a new environment variable to replace these called `PACTFLOW_DATABASE_AUTO_MIGRATE`, set its value to `true`. This environment variable will allow your existing database to be updated with the required structures and data to support Pactflow, and maintain any existing data.
```
But only if you had the previous two environment variables set.

Inksprout replied:
gotcha

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+
+1. Select one of the Pact Broker databases to update and use going forwards.
+2. Follow instructions above to set up Pactflow using this database.
+3. The Pact Broker instances that was used in the upgrade process and be decommissioned, as the database and it's contents are now available in Pactflow.
```
its (no comma)

mefellows replied:
![Screen Shot 2022-08-17 at 2 31 36 pm](https://user-images.githubusercontent.com/53900/185035152-892d3a46-c7f4-49ed-ae24-238ee0c3ab1f.png)

It's an apostrophe - ha! Got you grammar pedant!

bethesque replied:
Can grammar, just can't word 😆

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+1. Select one of the Pact Broker databases to update and use going forwards.
+2. Follow instructions above to set up Pactflow using this database.
+3. The Pact Broker instances that was used in the upgrade process and be decommissioned, as the database and it's contents are now available in Pactflow.
+4. Run the Pactflow instance and any other existing Pact Broker instances in parallel. You will need to update API calls such as publishing pacts and checking can-i-deploy to utilise both Pactflow and the old Pact Brokers. In this way all new data can gradually added to Pactflow while ensuring `can-i-deploy` still has access to production versions of all services.
```
This could use more detail I think. "Duplicate the pact publication and pact verification tasks in your pipeline. The second task should use the new Pactflow instance details."

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,39 @@
+2. Follow instructions above to set up Pactflow using this database.
+3. The Pact Broker instances that was used in the upgrade process and be decommissioned, as the database and it's contents are now available in Pactflow.
+4. Run the Pactflow instance and any other existing Pact Broker instances in parallel. You will need to update API calls such as publishing pacts and checking can-i-deploy to utilise both Pactflow and the old Pact Brokers. In this way all new data can gradually added to Pactflow while ensuring `can-i-deploy` still has access to production versions of all services.
+5. Continue to run Pactflow and Pact Brokers in parallel. Once the Pactflow instance contains all the `production` versions for all the services used in the contract tests the old Pact Broker instances can be safely decommissioned. This will allow Pactflow to be the soul source of data for 'can-i-deploy'
```
soul -> sole.

bethesque commented:
Awesome work Candy.

Inksprout commented:
Have addressed feedback. Also changed some of the note sections to tip/info as the note format is very plain.
![image](https://user-images.githubusercontent.com/13867336/185288832-10e4947b-fd84-4ff0-b9cc-9521f7a7f08a.png)
![image](https://user-images.githubusercontent.com/13867336/185289327-0f25993e-0a82-4bfc-b88e-c5bb4388bcfa.png)

![image](https://user-images.githubusercontent.com/13867336/185289132-cf6db072-f0ca-4f9b-849c-b2c529ea5afd.png)

bethesque commented on website/docs/docs/getting-started.mdx:
Just to be SUPER clear, because users are dumb, how about `and ant to migrate to the SaaS platform?`

Inksprout replied:
and ant?

Inksprout replied:
oh and want

bethesque replied:
AND ANT!!! AND ANT!!! 🐜

Inksprout replied:
don't forget to migrate your ants 🐜 🐜 🐜 🐜 🐜

bethesque commented on website/docs/docs/on-premises/database.md:
```diff
@@ -4,6 +4,10 @@ title: Database
 A PostgreSQL database is required for storage of the application data.
 
+:::info
+ If you would like to migrate an existing PostgreSQL database, skip this section and instead follow the guide [**here**](https://docs.pactflow.io/docs/on-premises/installation/migrating).
```
It's best to use URLs without the host in them, so they can be tested locally, so just keep from `/docs/` on.

bethesque replied:
Ditto for all the other URLs.

bethesque commented on website/docs/docs/on-premises/installation/migrating-from-pact-broker.md:
```diff
@@ -0,0 +1,43 @@
+
+If you have previously used the open source Pact Broker with a PostgreSQL database it is possible to migrate that data to Pactflow, keeping everything and making it available in Pactflow. The migration process uses the existing Pact Broker database. Database migrations are then performed to add additional tables and data used for Pactflow features.
+
+If there is no existing Pact Broker instance continue to the [Database](https://docs.pactflow.io/docs/on-premises/database) section.
```
I'm confused about this line. If there is no existing Pact Broker instance, then they wouldn't be in the "Migrating from Pact Broker" docs, yeah?

Inksprout replied:
trying to add clarity for people who are just going down the list in this on-premises -> Installation section
![image](https://user-images.githubusercontent.com/13867336/185295927-6dd5bb80-09fe-42b6-9e7b-3eb61ce5f957.png)
Maybe the wrong section for this doc to be in not sure

bethesque commented:
Shipit!
